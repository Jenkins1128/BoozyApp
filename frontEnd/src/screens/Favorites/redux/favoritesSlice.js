import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
