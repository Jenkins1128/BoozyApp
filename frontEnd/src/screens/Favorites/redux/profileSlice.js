import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
