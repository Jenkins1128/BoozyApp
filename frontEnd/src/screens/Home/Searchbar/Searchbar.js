import React from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet, TouchableWithoutFeedback } from 'react-native';
import { Feather, MaterialIcons } from '@expo/vector-icons';

const Searchbar = ({ state, dispatch, updateLocation, showFilterOverlay, getDataFromFilter, dismiss }) => {
	return (
		<TouchableWithoutFeedback onPress={dismiss}>
			<View style={styles.searchContainer}>
				<View style={styles.searchNearbyTextContainer}>
					<Text style={styles.searchNearbyText}>Search for happy hours nearby.</Text>
				</View>
				<View style={styles.searchBarBackground}>
					<Feather name='search' style={styles.searchIconStyle} />
					<TextInput
						style={styles.inputStyle}
						multiline={false}
						returnKeyType='next'
						onKeyPress={(ev) => {
							if (ev.nativeEvent.key == 'Enter') {
								getDataFromFilter();
							}
						}}
						onChangeText={(location) => dispatch(updateLocation({ location: location }))}
						value={state.location}
						placeholder='Enter City or Zip code'
					/>
					<TouchableOpacity onPress={showFilterOverlay}>
						<MaterialIcons name='filter-list' style={styles.filterIconStyle} />
					</TouchableOpacity>
				</View>
				<TouchableOpacity onPress={getDataFromFilter} style={styles.searchButton}>
					<Text style={styles.searchtext}>search</Text>
				</TouchableOpacity>
			</View>
		</TouchableWithoutFeedback>
	);
};

export default Searchbar;

const styles = StyleSheet.create({
	searchContainer: {
		flex: 1,
		width: '100%',
		padding: 20
	},
	searchNearbyTextContainer: {
		marginBottom: 10
	},
	searchNearbyText: {
		color: '#EB8873',
		fontFamily: 'Arial',
		fontWeight: 'bold',
		fontSize: 18
	},
	searchBarBackground: {
		backgroundColor: '#F0EEEE',
		height: 50,
		width: '90%',
		borderRadius: 15,
		alignSelf: 'center',
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'space-between',
		paddingHorizontal: 20
	},
	searchIconStyle: {
		fontSize: 35,
		color: '#E91E63'
	},
	filterIconStyle: {
		fontSize: 35,
		color: '#E91E63'
	},
	inputStyle: {
		fontSize: 18
	},
	searchButton: {
		alignSelf: 'flex-end',
		marginTop: 5,
		marginRight: 20,
		color: '#E91E63'
	},

	searchtext: {
		color: '#E91E63'
	}
});
