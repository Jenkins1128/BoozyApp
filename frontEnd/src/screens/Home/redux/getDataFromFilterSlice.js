import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const getDataFromFilterAsync = createAsyncThunk('getDataFromFilterAsync/status', async (data, { rejectWithValue }) => {
	const { location, priceType, cuisine } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${location}/${priceType}/${cuisine}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getDataFromFilterSlice = createSlice({
	name: 'getDataFromFilterAsync',
	initialState,
	reducers: {
		resetDataFromFilterRequestStatus: (state) => {
			state.currentState.dataFromFilterRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getDataFromFilterAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				dataFromFilterRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getDataFromFilterAsync.rejected, (state) => {
			state.currentState.dataFromFilterRequestStatus = 'rejected';
		});
	}
});

export const { resetDataFromFilterRequestStatus } = getDataFromFilterSlice.actions;
export const selectDataFromFilterState = (state) => state.dataFromFilter.currentState;
export default getDataFromFilterSlice.reducer;
