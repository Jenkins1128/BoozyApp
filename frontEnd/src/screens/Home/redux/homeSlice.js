import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {
		location: '',
		cuisine: null,
		priceType: 6,
		$color: 'white',
		$$color: 'white',
		$$$color: 'white',
		$$$$color: 'white',
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: [],
		initialPosition: {
			latitude: 0,
			longitude: 0
		},
		markerPosition: {
			latitude: 0,
			longitude: 0
		}
	}
};

export const getCurrentLocationDataAsync = createAsyncThunk('getCurrentLocationDataAsync/status', async (data, { rejectWithValue }) => {
	const { latitude, longitude } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${latitude}/${longitude}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const homeSlice = createSlice({
	name: 'getCurrentLocationDataAsync',
	initialState,
	reducers: {
		updateLocation: (state, { payload }) => {
			state.currentState.location = payload.location;
		},
		updateCuisine: (state, { payload }) => {
			state.currentState.cuisine = payload.cuisine;
		},
		updatePriceType: (state, { payload }) => {
			state.currentState.priceType = payload.priceType;
		},
		update$color: (state, { payload }) => {
			state.currentState.$color = payload.$color;
		},
		update$$color: (state, { payload }) => {
			state.currentState.$$color = payload.$$color;
		},
		update$$$color: (state, { payload }) => {
			state.currentState.$$$color = payload.$$$color;
		},
		update$$$$color: (state, { payload }) => {
			state.currentState.$$$$color = payload.$$$$color;
		},
		updateAllCategories: (state, { payload }) => {
			state.currentState.allCategories = payload.allCategories;
		},
		updateAlreadyFavorited: (state, { payload }) => {
			state.currentState.alreadyFavorited = payload.alreadyFavorited;
		},
		updateAlreadyRated: (state, { payload }) => {
			state.currentState.alreadyRated = payload.alreadyRated;
		},
		updateRestaurantsArray: (state, { payload }) => {
			state.currentState.restaurantsArray = payload.restaurantsArray;
		},
		updateInitialPosition: (state, { payload }) => {
			state.currentState.initialPosition = payload.initialPosition;
		},
		updateMarkerPosition: (state, { payload }) => {
			state.currentState.markerPosition = payload.markerPosition;
		},
		reset: (state) => {
			Object.assign(state.currentState, {
				location: '',
				cuisine: null,
				priceType: 6,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice1: (state) => {
			Object.assign(state.currentState, {
				priceType: 1,
				$color: 'yellow',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice2: (state) => {
			Object.assign(state.currentState, {
				priceType: 2,
				$color: 'white',
				$$color: 'yellow',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice3: (state) => {
			Object.assign(state.currentState, {
				priceType: 3,
				$color: 'white',
				$$color: 'white',
				$$$color: 'yellow',
				$$$$color: 'white'
			});
		},
		setPrice4: (state) => {
			Object.assign(state.currentState, {
				priceType: 4,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'yellow'
			});
		},
		resetHomeRequestStatus: (state) => {
			state.currentState.locationDataRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getCurrentLocationDataAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					locationDataRequestStatus: 'fulfilled',
					restaurantsArray: payload
				});
			})
			.addCase(getCurrentLocationDataAsync.rejected, (state) => {
				state.currentState.locationDataRequestStatus = 'rejected';
			});
	}
});

export const {
	updateLocation,
	updateCuisine,
	updatePriceType,
	update$color,
	update$$color,
	update$$$color,
	update$$$$color,
	updateAllCategories,
	updateAlreadyFavorited,
	updateAlreadyRated,
	updateRestaurantsArray,
	updateInitialPosition,
	updateMarkerPosition,
	reset,
	setPrice1,
	setPrice2,
	setPrice3,
	setPrice4,
	resetHomeRequestStatus
} = homeSlice.actions;

export const selectHomeState = (state) => state.home.currentState;
export default homeSlice.reducer;
