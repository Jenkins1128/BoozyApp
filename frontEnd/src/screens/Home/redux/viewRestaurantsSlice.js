import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const viewRestaurantsAsync = createAsyncThunk('viewRestaurantsAsync/status', async (data, { rejectWithValue }) => {
	const { index, state } = data;
	state.index = index;
	let restaurantId;
	if (state.restaurantsArray[index]['id']) {
		restaurantId = state.restaurantsArray[index]['id'];
	} else if (state.restaurantsArray[index]['restaurantId']) {
		restaurantId = state.restaurantsArray[index]['restaurantId'];
	}
	const name = state.restaurantsArray[index]['name'];
	try {
		const alreadyFavoritedResponse = await axios.post(`${baseUrl}/favorites`, {
			restaurantId: restaurantId,
			name: name
		});
		const alreadyRatedResponse = await axios.get(`${baseUrl}/${restaurantId}/review`);
		const categories = state.restaurantsArray[index]['categories'] ? state.restaurantsArray[index]['categories'].map((obj) => obj.title) : [];
		const categoriesString = categories.length ? categories.join(', ') : '';
		const newState = {
			...state,
			alreadyFavorited: alreadyFavoritedResponse.data[0]['contains'],
			alreadyRated: alreadyRatedResponse.data[0]['contains'],
			allCategories: categoriesString,
			restaurantsArray: [state.restaurantsArray[index]]
		};
		return newState;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const viewRestaurantsSlice = createSlice({
	name: 'viewRestaurantsAsync',
	initialState,
	reducers: {
		resetRestaurantRequestStatus: (state) => {
			state.currentState.restaurantRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(viewRestaurantsAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					restaurantInfo: payload,
					restaurantRequestStatus: 'fulfilled'
				});
			})
			.addCase(viewRestaurantsAsync.rejected, (state) => {
				state.currentState.restaurantRequestStatus = 'rejected';
			});
	}
});

export const { resetRestaurantRequestStatus } = viewRestaurantsSlice.actions;
export const selectRestaurantState = (state) => state.restaurant.currentState;

export default viewRestaurantsSlice.reducer;
