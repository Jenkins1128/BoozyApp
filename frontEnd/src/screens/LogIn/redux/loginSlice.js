import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const loginAsync = createAsyncThunk('loginAsync/status', async (data, { rejectWithValue }) => {
	const { email, password } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/login', {
			email,
			password
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const loginSlice = createSlice({
	name: 'loginAsync',
	initialState,
	reducers: {
		updateEmail: (state, { payload }) => {
			state.currentState.email = payload.email;
		},
		updatePassword: (state, { payload }) => {
			state.currentState.password = payload.password;
		},
		resetStatus: (state) => {
			state.currentState.requestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(loginAsync.pending, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'pending',
					isLoading: true
				});
			})
			.addCase(loginAsync.fulfilled, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'fulfilled',
					isLoading: false
				});
			})
			.addCase(loginAsync.rejected, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'rejected',
					isLoading: false
				});
			});
	}
});

export const { updateEmail, updatePassword, resetStatus } = loginSlice.actions;
export const selectLogInState = (state) => state.login.currentState;

export default loginSlice.reducer;
