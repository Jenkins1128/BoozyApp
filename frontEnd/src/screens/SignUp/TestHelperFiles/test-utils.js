import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
