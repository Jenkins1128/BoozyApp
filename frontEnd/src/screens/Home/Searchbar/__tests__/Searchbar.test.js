import React from 'react';
import Searchbar from '../Searchbar';
import { cleanup, render, fireEvent } from '../../../../TestHelperFiles/redux/test-utils';

describe('<Searchbar />', () => {
	let state;
	let updateLocation, dispatch, showFilterOverlay, getDataFromFilter;
	let rendered;

	beforeEach(() => {
		state = {
			location: 'France'
		};
		(updateLocation = jest.fn()), (dispatch = jest.fn()), (showFilterOverlay = jest.fn()), (getDataFromFilter = jest.fn());
		rendered = render(<Searchbar state={state} updateLocation={updateLocation} dispatch={dispatch} showFilterOverlay={showFilterOverlay} getDataFromFilter={getDataFromFilter} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap view with a flexible wrapper', () => {
		const containerViewComponent = rendered.getByTestId('searchContainer');
		const given = containerViewComponent.props.style;
		const result = {
			flex: 1,
			width: '100%',
			padding: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should wrap view with a flexible wrapper', () => {
		const containerViewComponent = rendered.getByTestId('searchContainer');
		const given = containerViewComponent.props.style;
		const result = {
			flex: 1,
			width: '100%',
			padding: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render searchNearbyTextContainer styles correctly', () => {
		const searchNearbyTextContainerComponent = rendered.getByTestId('searchNearbyTextContainer');
		const given = searchNearbyTextContainerComponent.props.style;
		const result = {
			marginBottom: 10
		};
		expect(result).toMatchObject(given);
	});

	it('should render searchNearbyText styles correctly', () => {
		const searchNearbyTextComponent = rendered.getByTestId('searchNearbyText');
		const given = searchNearbyTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontFamily: 'Arial',
			fontWeight: 'bold',
			fontSize: 18
		};
		expect(result).toMatchObject(given);
	});

	it('should render searchBarBackground styles correctly', () => {
		const searchBarBackgroundComponent = rendered.getByTestId('searchBarBackground');
		const given = searchBarBackgroundComponent.props.style;
		const result = {
			backgroundColor: '#F0EEEE',
			height: 50,
			width: '90%',
			borderRadius: 15,
			alignSelf: 'center',
			flexDirection: 'row',
			alignItems: 'center',
			justifyContent: 'space-between',
			paddingHorizontal: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render searchInput styles correctly', () => {
		const searchInputComponent = rendered.getByTestId('searchInput');
		const given = searchInputComponent.props.style;
		const result = {
			fontSize: 18
		};
		expect(result).toMatchObject(given);
	});

	it('should render searchButton correctly', () => {
		const searchButtonComponent = rendered.getByTestId('searchButton');
		const given = searchButtonComponent.props.style;
		const result = {
			alignSelf: 'flex-end',
			marginTop: 5,
			marginRight: 20,
			color: '#E91E63',
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render searchtext correctly', () => {
		const containerViewComponent = rendered.getByTestId('searchtext');
		const given = containerViewComponent.props.style;
		const result = {
			color: '#E91E63'
		};
		expect(result).toMatchObject(given);
	});

	//interaction
	it('should fire searchInput onChangeText event', () => {
		const searchInputComponent = rendered.getByTestId('searchInput');
		fireEvent(searchInputComponent, 'changeText', 'new text');
		expect(updateLocation).toHaveBeenCalledWith({ location: 'new text' });
	});

	it('should fire search button event', () => {
		const getDataFromFilterButton = rendered.getByText('search');
		fireEvent.press(getDataFromFilterButton);
		expect(getDataFromFilter).toHaveBeenCalledTimes(1);
	});

	it('should fire show filter button event', () => {
		const showFilterButton = rendered.getByTestId('showFilterButton');
		fireEvent.press(showFilterButton);
		expect(showFilterOverlay).toHaveBeenCalledTimes(1);
	});
});
