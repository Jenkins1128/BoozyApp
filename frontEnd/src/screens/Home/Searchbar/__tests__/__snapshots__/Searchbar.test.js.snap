// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchbar /> should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "padding": 20,
      "width": "100%",
    }
  }
  testID="searchContainer"
>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
    testID="searchNearbyTextContainer"
  >
    <Text
      style={
        Object {
          "color": "#EB8873",
          "fontFamily": "Arial",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
      testID="searchNearbyText"
    >
      Search for happy hours nearby.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F0EEEE",
        "borderRadius": 15,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "width": "90%",
      }
    }
    testID="searchBarBackground"
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      multiline={false}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter City or Zip code"
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "fontSize": 18,
        }
      }
      testID="searchInput"
      underlineColorAndroid="transparent"
      value="France"
    />
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "color": "#E91E63",
        "marginRight": 20,
        "marginTop": 5,
        "opacity": 1,
      }
    }
    testID="searchButton"
  >
    <Text
      style={
        Object {
          "color": "#E91E63",
        }
      }
      testID="searchtext"
    >
      search
    </Text>
  </View>
</View>
`;
